### PR TITLE
godeps update

### PR DESCRIPTION
**Pull request by cryptix**
Motivation was the upstream fix of #956.

I updated `jbenet/go-reuseport` as well, which closes #943 for now. It was to big for one PR anyway.

This PR also contains large-ish upstream changes from the following deps:

* Sirupsen/logrus
* syndtr/goleveldb
* syndtr/gosnappy

and a couple of smaller ones.

Also removes `gorilla/context` and `gorilla/mux`. I assume they could have been dropped since #887.

### CODE REVIEW

File: Godeps/_workspace/src/bazil.org/fuse/syscallx/syscallx_std.go [hunk 1]
```diff
@@ -3,24 +3,24 @@
 package syscallx
 
 // This file just contains wrappers for platforms that already have
-// the right stuff in stdlib.
+// the right stuff in golang.org/x/sys/unix.
 
 import (
-	"syscall"
+	"golang.org/x/sys/unix"
```
Comment (cryptix) [inline Godeps/_workspace/src/bazil.org/fuse/syscallx/syscallx_std.go]: no idea why 'godep save' didn't pick up this one... 

cc @jbenet - wait with merge until this is fixed

Comment (cryptix) [inline Godeps/_workspace/src/bazil.org/fuse/syscallx/syscallx_std.go]: I'm even more confused now..

I cloned this branch into an empty GOPATH on two non-darwin machines and it builds fine..
Am I missing something about this `syscallx` package?

cc @tv42

Comment (tv42) [inline Godeps/_workspace/src/bazil.org/fuse/syscallx/syscallx_std.go]: @cryptix Not sure what you're talking about. That diff looks like commit 3c90b2396ff41bbe52a7b58b606d75cf47610a68 in bazil.org/fuse.

Comment (cryptix) [inline Godeps/_workspace/src/bazil.org/fuse/syscallx/syscallx_std.go]: @tv42  I was confused that `godep` didn't vendor `golang.org/x/sys/unix` and didn't rewrite the import path here to `prefix/_workspace/src/...`

My (way too nebulous) question for you was if the `bazil.org/fuse/fs -> bazil.org/fuse/syscallx -> golang.org/x/sys/unix` import chain is somehow conditional. I could have checked this myself this morning - `syscallx` is only included in the __tests__ of `bazil.org/fuse/fs` so godep is correct (it doesn't respect test deps at all) with not including it. 

My brain was only faking higher states of awareness, I guess. Sorry for the noise.

@jbenet we can move forward.

**Comment by jbenet** [comment]
So many deps changing in one PR! LGTM

**Pull request merged**